learning_resources_search/tasks.py

```diff
@@ -596,3 +706,22 @@ def finish_recreate_index(results, backing_indices):
         except RequestError as ex:
             raise RetryError(str(ex)) from ex
     log.info("recreate_index has finished successfully!")
+
+
+@app.task(
+    acks_late=True,
+    reject_on_worker_lost=True,
+    rate_limit=settings.NOTIFICATION_ATTEMPT_RATE_LIMIT,
+)
+def attempt_send_digest_email_batch(user_ids, template_data):
+    for user_id in user_ids:
+        user = User.objects.get(id=user_id)
+        total_count = sum(
+            [len(template_data[user_id][group]) for group in template_data[user_id]]
```

Comment:
Throwing an error when I run
```python
from learning_resources_search.tasks import send_subscription_emails
send_subscription_emails.delay('channel_subscription_type')
```


```python
celery-1                 |   File "/src/learning_resources_search/tasks.py", line 720, in attempt_send_digest_email_batch
celery-1                 |     [len(template_data[user_id][group]) for group in template_data[user_id]]
celery-1                 |                                                      ~~~~~~~~~~~~~^^^^^^^^^
celery-1                 | KeyError: 1

```

These changes work:

```python
@app.task(bind=True)
def send_subscription_emails(self, subscription_type, period="daily"):
....
    email_tasks = celery.group(
        [
            attempt_send_digest_email_batch.si(user_template_items)
            for user_template_items in chunks(
                template_data.items(), chunk_size=settings.NOTIFICATION_ATTEMPT_CHUNK_SIZE
            )
        ]
    )


@app.task(
    acks_late=True,
    reject_on_worker_lost=True,
    rate_limit=settings.NOTIFICATION_ATTEMPT_RATE_LIMIT,
)
def attempt_send_digest_email_batch(user_template_items):
    for (user_id, template_data) in user_template_items:
        log.info("Sending email to user %s", user_id)
        user = User.objects.get(id=user_id)
        total_count = sum(
            [len(template_data[group]) for group in template_data]
        )
        send_template_email(
            [user.email],
            f"{settings.MITOPEN_TITLE} New Learning Resources for You",
            "email/subscribed_channel_digest.html",
            context={"documents": template_data, "total_count": total_count},
        )
```

Unit tests for these functions should also be adjusted accordingly.